KubeArmor/kvmAgent/kvmAgent.go

```diff
@@ -53,9 +59,22 @@ func NewKVMAgent(eventCb tp.KubeArmorHostPolicyEventCallback) *KVMAgent {
 
 	kvm.gRPCServer = gRPCServer
 
+	cp := x509.NewCertPool()
+	if !cp.AppendCertsFromPEM(kvm.caCert) {
+		kg.Err("credentials: failed to append certificates")
+		return nil
+	}
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: false,
+		RootCAs:            cp,
+		ServerName:         "0.0.0.0",
+		MinVersion:         tls.VersionTLS12,
+	}
+
 	// Connect to gRPC server
 	ctx, _ := context.WithTimeout(context.Background(), time.Second*5)
-	gRPCConnection, err := grpc.DialContext(ctx, kvm.gRPCServer, grpc.WithInsecure(), grpc.WithBlock())
+	gRPCConnection, err := grpc.DialContext(ctx, kvm.gRPCServer, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)), grpc.WithBlock())
```

Comment:
Can we add the case if certs do not exist as exception handling?
In this implementation, it seems that KubeArmor will fail if no certs.
I feel when people test KubeArmor locally, they may not have any certs.

Reply:
Yes @nam-jaehyun. It is possible. Is it like when certs are not available, we have to establish connection in insecure mode ?
We can use InSecureSkipVerify in TLSConfig if certs are not available or we can use grpc.WithInSecure() option in Dial.